crates/ide-assists/src/tests.rs

```diff
@@ -346,6 +368,41 @@ fn labels(assists: &[Assist]) -> String {
     labels.into_iter().collect::<String>()
 }
 
+#[test]
+fn long_groups_are_skipped_under_skip_resolve_strategy() {
+    let before = r#"
+trait SomeTrait {
+    type T;
+    fn fn_(arg: u32) -> u32;
+    fn method_(&mut self) -> bool;
+}
+struct A;
+impl SomeTrait for A {
+    type T = u32;
+
+    fn fn_(arg: u32) -> u32 {
+        42
+    }
+
+    fn method_(&mut self) -> bool {
+        false
+    }
+}
+struct B {
+    a$0 : A,
+}
+    "#;
+    let (before_cursor_pos, before) = extract_offset(before);
+    let (db, file_id) = with_single_file(&before);
+    let frange = FileRange { file_id, range: TextRange::empty(before_cursor_pos) };
+    let res = assists(&db, &TEST_CONFIG, AssistResolveStrategy::None, frange.into());
```

Comment:
I think we have a `check_assist_not_applicable()` for that. It doesn't take a config, but it already has overloads for configs, so it feels fine to duplicate it once more.